src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -125,6 +131,39 @@ public boolean equals(Object obj)
         }
     }
 
+    static class BufferFuture
```

Comment:
Nit: Calling it a `BufferFuture` is a bit confusing to me, because this class is not a `Future`. Its purpose is also not even to provide the buffer. Its purpose is to load the contents into an existing buffer it wraps.

Reply:
Renamed to LoadingBuffer, unfortunately we already have BufferHolder :-)